### PR TITLE
Maya padding 5 overlap

### PR DESCRIPTION
**Pull request by mayalevy**
in that PR we:
1. increase min_event_size from 8 to 30
2. added cohortMinSize filter in R code to be set anyway. before that it was set only when min_support_filters=true (which was false by default)
is_too_small_event which decides the cohortMinSize was created to find only bps, now it supports be as well 
3. sum_qual: in case there is no MATE_ID in the input vcf, the qual was set for each be separately. Now, in cases that R linked them, it is set as the sum of qual of both of the variants. (only in case there was no MATEIS and R linked them).
we treat them as they have MATEID and hence, the qual threshold is lower
In addition, we set the filter to be the union of filters of both sides. With the exception that in case we have PASS and cohortMinSize, the union will be PASS.
6. In revert_sup_low_mapq_ua_alignment scrip, we revert as well primary aligned with low qual which have a pair which is supplementary aligned.
7. We added requiredReadAndAssemblyBreakpointOverlap to gridss overlapping of assembly with read. This means that for consider it as overlap we should have at least this number of overlapping bases
8. In gridss: CRAM support + no need to copy input to sv

### CODE REVIEW

File: scripts/revert_sup_low_mapq_ua_alignment.py [hunk 1]
```diff
@@ -17,14 +17,27 @@
...
+        sup_reads.add(after_read.query_name)
+
+# reads that should be reverted:
+# reads with supplementary alignment and has low mapping quality
+# reads with decoy alignment
+# reads which their primary mapping quality is low, and they have supplementary alignment read
+for after_read in after.fetch():
+        if (after_read.mapping_quality < args.min_mapping_quality and after_read.flag & 2048) or \
+            (after_read.has_tag('SA') and 'decoy' in after_read.get_tag('SA') or \
+            (after_read.mapping_quality < args.min_mapping_quality and after_read.flag == 0 and after_read.query_name in sup_reads)):
```
Comment (ilyasoifer) [inline scripts/revert_sup_low_mapq_ua_alignment.py]: can you explain what after_read.flag == 0 supposed to do?

File: scripts/link_breakpoints [hunk 2]
```diff
@@ -256,9 +260,10 @@ event_link_df = bind_rows(
   mutate(
     max_supporting_fragment_count = max(supporting_fragment_count),
     min_supporting_fragment_count = min(supporting_fragment_count),
-    hasPolyA=any(hasPolyA)
-    ) %>%
+    hasPolyA=any(hasPolyA),
+    sum_qual = if(any(has_pair, na.rm = TRUE)) { sum(QUAL, na.rm = TRUE) } else { 0 }) %>%
```
Comment (ilyasoifer) [inline scripts/link_breakpoints:265]: Is the sum happening on the variants that were linked before the script? It probably should not, the qualities are effectively aggregated there in gridss

**Review by ilyasoifer — COMMENTED** [review]
Hi - see two comments. 
Please provide summary of changes in the PR description, because this pull request covers very wide range of changes

File: scripts/revert_sup_low_mapq_ua_alignment.py (hunk 1 above)
Comment (mayalevy) [reply scripts/revert_sup_low_mapq_ua_alignment.py]: it means the variant is primary.
0 - primary
2048 - supplementary
I will explain in the code as well

File: scripts/link_breakpoints (hunk 2 above)
Comment (mayalevy) [reply scripts/link_breakpoints:265]: oh you are right,
 I had this logic, but I don't find that. I will add that, thanks

Comment (mayalevy) [reply scripts/link_breakpoints:265]: I have added that check tot he code, but it does not have any influence. All the variants in that table are without MATEID, so I think we will get the same results

File: scripts/revert_sup_low_mapq_ua_alignment.py (hunk 1 above)
Comment (ilyasoifer) [reply scripts/revert_sup_low_mapq_ua_alignment.py]: I don't think it works this way. SAM flag is bitwise: x&2048 is supplementary. I think the check for primary is ((x & 2048) == 0), but pysam api also has `not x.is_supplementary` I think

**Review by ilyasoifer — COMMENTED** [review]
just addingthis because I was not sure you saw my last comment

**Pull request merged**